plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2099,14 +2103,25 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                     bucketing.getColumns().stream()
                             .map(ColumnHandle.class::cast)
                             .collect(toImmutableList())));
+
+            if (isOptimizeSortedFileReads(session)) {
+                Map<String, ColumnHandle> columnHandles = getColumnHandles(session, table);
+                sortingProperties = hiveTable.getBucketHandle().map(HiveBucketHandle::getSortedBy)
+                        .orElse(ImmutableList.of())
+                        .stream()
+                        .map(sortingColumn -> new SortingProperty<>(
+                                columnHandles.get(sortingColumn.getColumnName()),
+                                sortingColumn.getOrder().getSortOrder()))
+                        .collect(toImmutableList());
+            }
         }
 
         return new ConnectorTableProperties(
                 predicate,
                 tablePartitioning,
                 Optional.empty(),
                 discretePredicates,
-                ImmutableList.of());
+                sortingProperties);
```

Comment:
Also, since the point of this change is to let planner derived a better plan for such tables, can we have an `assertPlan`-based test for this? Ideally if this is added in a prep commit, so that the commit diff shows also how the plan is improved by this change.

Reply:
I've added a commit to this PR which avoids partial sort with pre-sorted files and plan assertions for it